app/hooks/use-price-conversion.ts

```diff
@@ -0,0 +1,69 @@
+import { usePriceContext } from "@app/store/price-context"
+import { PaymentAmount, WalletCurrency } from "@app/types/amounts"
+import * as React from "react"
+
+export const usePriceConversion = () => {
+  const priceData = usePriceContext()
+  const convertCurrencyAmount = React.useCallback(
```

Comment:
I know this isn't your code but could we add types for args and return value for this function?